x-pack/plugins/stack_alerts/public/alert_types/components/index_select_popover.tsx

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { isString } from 'lodash';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiButtonIcon,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiExpression,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiPopover,
+  EuiPopoverTitle,
+  EuiSelect,
+} from '@elastic/eui';
+import { HttpSetup } from 'kibana/public';
+import { useKibana } from '../../../../../../src/plugins/kibana_react/public';
+import {
+  firstFieldOption,
+  getFields,
+  getIndexOptions,
+  getIndexPatterns,
+  getTimeFieldOptions,
+  IErrorObject,
+} from '../../../../triggers_actions_ui/public';
+
+interface KibanaDeps {
+  http: HttpSetup;
+}
+
+interface Props {
+  index: string[];
+  esFields: Array<{
+    name: string;
+    type: string;
+    normalizedType: string;
+    searchable: boolean;
+    aggregatable: boolean;
+  }>;
+  timeField: string | undefined;
+  errors: IErrorObject;
+  onIndexChange: (indices: string[]) => void;
+  onTimeFieldChange: (timeField: string) => void;
+}
+
+export const IndexSelectPopover: React.FunctionComponent<Props> = ({
+  index,
+  esFields,
+  timeField,
+  errors,
+  onIndexChange,
+  onTimeFieldChange,
+}) => {
+  const { http } = useKibana<KibanaDeps>().services;
+
+  const [indexPopoverOpen, setIndexPopoverOpen] = useState(false);
+  const [indexOptions, setIndexOptions] = useState<EuiComboBoxOptionOption[]>([]);
+  const [indexPatterns, setIndexPatterns] = useState([]);
+  const [areIndicesLoading, setAreIndicesLoading] = useState<boolean>(false);
+  const [timeFieldOptions, setTimeFieldOptions] = useState([firstFieldOption]);
+
+  useEffect(() => {
+    const indexPatternsFunction = async () => {
+      setIndexPatterns(await getIndexPatterns());
+    };
+    indexPatternsFunction();
+  }, []);
+
+  useEffect(() => {
+    const timeFields = getTimeFieldOptions(esFields);
+    setTimeFieldOptions([firstFieldOption, ...timeFields]);
+  }, [esFields]);
+
+  const renderIndices = (indices: string[]) => {
+    const rows = indices.map((s: string, i: number) => {
```

Comment:
nit.
can we rename `s` and `i` (`esIndex` and `index` perhaps?) to more meaningful names?
Sorry, pet peeve of mine 😬